src/dmd/root/ctfloat.d

```diff
@@ -229,4 +229,13 @@ extern (C++) struct CTFloat
         minusone = real_t(-1);
         half = real_t(0.5);
     }
+
+    /// Deinitalize the compile-time floating-point globals.
+    @trusted
+    static void deinitialize()
```

Comment:
Why is it called de-initialize when what it actually does is initialize?

Reply:
It doesn't do initialization. The initialization is done here:

https://github.com/dlang/dmd/blob/8d760396c9a5e663614cc81175806f89eb0b6225/src/dmd/root/ctfloat.d#L225-L231

Reply:
What it appears to do is initialize, and should be part of the initialization code.

Reply:
Please explain. It sets the values to their `init` values. This is **not** what the `initialize` function is doing.

Reply:
`resetStaticFields!(typeof(this));` will expand to this:

```d
zero = zero.init;
one = one.init;
minusone = minusone.init;
half = half.init;
```

Reply:
What purpose does that serve when the .init values serve no purpose?

Reply:
Why isn't, for example, `CTfloat.zero` declared as:
```
__gshared real_t zero = real_t(0);
```
which will be successful if the above works in CTFE, which it should. Then all this code can be removed.

Reply:
I can try that.

Reply:
Changed to:

```d
__gshared const real_t zero = 0;
```